src/components/Details/DetailDescription.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import { AppWorkload } from '../../types/App';
+import { Badge, PopoverPosition, Title, Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { style } from 'typestyle';
+import { Link } from 'react-router-dom';
+import MissingSidecar from '../MissingSidecar/MissingSidecar';
+import * as H from '../../types/Health';
+import { renderTrafficStatus } from '../Health/HealthDetails';
+import { createIcon } from '../Health/Helper';
+import { HealthSubItem } from '../../types/Health';
+
+type Props = {
+  namespace: string;
+  apps?: string[];
+  workloads?: AppWorkload[];
+  services?: string[];
+  health?: H.Health;
+};
+
+const iconStyle = style({
+  margin: '0 0 0 0',
+  padding: '0 0 0 0',
+  display: 'inline-block'
+});
+
+const resourceListStyle = style({
+  margin: '0px 0 11px 0'
+});
+
+const titleStyle = style({
+  margin: '15px 0 8px 0'
+});
+
+class DetailDescription extends React.PureComponent<Props> {
+  private renderAppItem(namespace: string, appName: string) {
+    return (
+      <li key={`App_${appName}`}>
+        <div key="service-icon" className={iconStyle}>
+          <Tooltip position={TooltipPosition.top} content={<>Application</>}>
+            <Badge className={'virtualitem_badge_definition'}>A</Badge>
+          </Tooltip>
+        </div>
+        <span>
+          <Link to={'/namespaces/' + namespace + '/applications/' + appName}>{appName}</Link>
+        </span>
+      </li>
+    );
+  }
+
+  private renderServiceItem(namespace: string, serviceName: string) {
+    return (
+      <li key={`Service_${serviceName}`}>
+        <div key="service-icon" className={iconStyle}>
+          <Tooltip position={TooltipPosition.top} content={<>Service</>}>
+            <Badge className={'virtualitem_badge_definition'}>S</Badge>
+          </Tooltip>
+        </div>
+        <span>
+          <Link to={'/namespaces/' + namespace + '/services/' + serviceName}>{serviceName}</Link>
+        </span>
+      </li>
+    );
+  }
+
+  private renderEmptyItem(type: string) {
+    const message = 'No ' + type + ' found';
+    return <div> {message} </div>;
+  }
+
+  private appList() {
+    const applicationList =
+      this.props.apps && this.props.apps.length > 0
+        ? this.props.apps.map(name => this.renderAppItem(this.props.namespace, name))
+        : this.renderEmptyItem('applications');
+
+    return [
+      <div key="service-list" className={resourceListStyle}>
+        <Title headingLevel="h5" size="lg" className={titleStyle}>
+          Applications
+        </Title>
+        <ul style={{ listStyleType: 'none' }}>{applicationList}</ul>
+      </div>
+    ];
+  }
+
+  private renderWorkloadItem(workload: AppWorkload) {
+    return (
+      <span>
+        <div key="service-icon" className={iconStyle}>
+          <Tooltip position={TooltipPosition.top} content={<>Workload</>}>
+            <Badge className={'virtualitem_badge_definition'}>W</Badge>
+          </Tooltip>
```

Comment:
Please replace this with the new PFBadge:

`<PFBadge badge={PFBadges.Workload} position={TooltipPosition.top} />`